lib/elixir/lib/string.ex

```diff
@@ -1812,4 +1812,104 @@ defmodule String do
       {comm + 1, trans, current}
     end
   end
+
+  @doc """
+  Returns a keyword list that represents an edit graph.
+
+  The algorithm is outlined in the "An O(ND) Difference Algorithm and Its Variations" paper.
+
+  ## Examples
+
+      iex> string1 = "fox hops over the dog"
+      iex> string2 = "fox jumps over the lazy cat"
+      iex> String.myers_difference(string1, string2)
+      [eq: "fox ", del: "ho", ins: "jum", eq: "ps over the ", del: "dog", ins: "lazy cat"]
+
+  """
+  @spec myers_difference(t, t) :: [{:eq | :ins | :del, t}] | nil
+  def myers_difference(str1, str2) do
+    {chars1, len1} = dissect(str1)
+    {chars2, len2} = dissect(str2)
+
+    path = {0, 0, chars1, chars2, []}
+    find_script(0, len1 + len2, [path])
+  end
+
+  defp find_script(envelope, max, _paths) when envelope > max do
+    nil
+  end
+
+  defp find_script(envelope, max, paths) do
+    case each_diagonal(-envelope, envelope, paths, []) do
+      {:done, edits} -> compact_reverse(edits, [])
+      {:next, paths} -> find_script(envelope + 1, max, paths)
+    end
+  end
+
+  defp compact_reverse([], acc), do: acc
+
+  defp compact_reverse([{kind, char} | rest], [{kind, chars} | acc]) do
+    compact_reverse(rest, [{kind, char <> chars} | acc])
+  end
+
+  defp compact_reverse([elem | rest], acc) do
+    compact_reverse(rest, [elem | acc])
+  end
+
+  defp each_diagonal(diag, limit, _paths, next_paths) when diag > limit do
+    {:next, Enum.reverse(next_paths)}
+  end
+
+  defp each_diagonal(diag, limit, paths, next_paths) do
+    {path, rest} = proceed_path(diag, limit, paths)
+    with {:cont, path} <- follow_snake(path) do
+      each_diagonal(diag + 2, limit, rest, [path | next_paths])
+    end
```

Comment:
@whatyouhide I'll stick with `with` as it quite the same if we put `{:done, edits}` (`result -> result`) as second pattern.